src/APIM_ARMTemplate/apimtemplate/Common/Exceptions/NoApiVersionSetWithSuchNameFoundException.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System;
+
+namespace apimtemplate.Common.Exceptions
+{
+    public class NoApiVersionSetWithSuchNameFoundException : Exception
```

Comment:
Are we catching this type of exception? If not, what is the added value of a dedicated type?

Reply:
We are not right now. It is done for future building library as a NuGet. It will be much better for end user to catch specific exception instead of general `Exception`.

Also we can test the behaviour in that case with specific exception, not just a general one. Consider this:
```
public void MyTest()
{
    var extractorExecutor = new ExtractorExecutor();
    var expectedException = Assert.Throws<Exception>(() => extractorExecutor.RunGeneration());
}
```

If `RunGeneration()` has an internal NRE, then test will pass, however it will not be an exception we were expecting to be thrown. That is why I prefer to dedicate a separate type - if an exception has occured, we need to specify an error as much as possible